src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/EventAudit.kt

```diff
@@ -1,58 +1,72 @@
 package uk.gov.justice.digital.hmpps.visitscheduler.model.entity
 
+import jakarta.persistence.CascadeType.REFRESH
 import jakarta.persistence.Column
 import jakarta.persistence.Entity
 import jakarta.persistence.EnumType
 import jakarta.persistence.Enumerated
+import jakarta.persistence.FetchType
 import jakarta.persistence.GeneratedValue
 import jakarta.persistence.GenerationType
 import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
 import jakarta.persistence.Table
 import org.hibernate.annotations.CreationTimestamp
 import uk.gov.justice.digital.hmpps.visitscheduler.dto.enums.ApplicationMethodType
 import uk.gov.justice.digital.hmpps.visitscheduler.dto.enums.EventAuditType
-import uk.gov.justice.digital.hmpps.visitscheduler.dto.enums.UserType
 import java.time.LocalDateTime
 
 @Entity
 @Table(
   name = "event_audit",
 )
-class EventAudit(
+class EventAudit private constructor(
 
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   @Column(name = "ID")
   val id: Long = 0,
 
   @Column
-  var bookingReference: String? = null,
+  val bookingReference: String? = null,
 
   @Column
-  var applicationReference: String? = null,
+  val applicationReference: String? = null,
 
   @Column
-  var sessionTemplateReference: String? = null,
+  val sessionTemplateReference: String? = null,
 
   @Column(nullable = false)
   @Enumerated(EnumType.STRING)
-  var type: EventAuditType,
+  val type: EventAuditType,
 
   @Column(nullable = false)
   @Enumerated(EnumType.STRING)
-  var applicationMethodType: ApplicationMethodType,
+  val applicationMethodType: ApplicationMethodType,
 
-  @Column(nullable = false)
-  var actionedBy: String,
+  @Column(name = "ACTIONED_BY_ID", nullable = true)
+  private val actionedById: Long? = null,
```

Comment:
If the actionedBy is a SYSTEM_USER, would it not have an actionedById?

Reply:
Basically, does this need to be nullable?

Reply:
needs to be nullable = false,.... I think?

Reply:
Yes I think so.
PUBLIC / STAFF / SYSTEM, all of these will contain an Actioned_by_id, with different columns within the ActionedBy table being null...

But the ID field is mandatory right?